datasets/hate_offensive/hate_offensive.py

```diff
@@ -0,0 +1,93 @@
+
+_HOMEPAGE = "https://arxiv.org/abs/1905.12516"
+
+_LICENSE = "https://github.com/t-davidson/hate-speech-and-offensive-language/blob/master/LICENSE"
+
+_URLs = "https://github.com/t-davidson/hate-speech-and-offensive-language/raw/master/data/labeled_data.csv"
+
+
+class HateOffensive(datasets.GeneratorBasedBuilder):
+    """Automated Hate Speech Detection and the Problem of Offensive Language """
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "count": datasets.Value("int32"),
+                "hate_speech": datasets.Value("int32"),
+                "offensive_language": datasets.Value("int32"),
+                "neither": datasets.Value("int32"),
+                "class": datasets.ClassLabel(
+                    names=["0", "1", "2"]
+                ),  # 0 - hate speech 1 - offensive language 2 - neither
+                "tweet": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": os.path.join(data_dir)},
+            )
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+        f = pd.read_csv(filepath)
+        for id_, row in f.iterrows():
+            yield id_, {
+                "count": row["count"],
+                "hate_speech": row["hate_speech"],
+                "offensive_language": row["offensive_language"],
+                "neither": row["neither"],
+                "class": row["class"],
+                "tweet": str(row["tweet"]),
+            }
```

Comment:
Could you re-writethis using the `csv` package? It would avoid loading an additional dependency :)

See the docs here: https://docs.python.org/fr/3.6/library/csv.html